File: src/Java/1. First Course/5. Fourth Hole/src/main/refactoringgolf/store/Employee.java
package refactoringgolf.store;


public abstract class Employee {

	protected String firstName;
	protected String lastName;
	protected float fixedSalary;
	protected Employee manager;
	protected String street;
	protected String city;
	protected String country;
	protected Employee(String firstName, String lastName, float fixedSalary) {
		this.firstName = firstName;
		this.lastName = lastName;
		this.fixedSalary = fixedSalary;
	}

	public String getFirstName() {
		return firstName;
	}

	public String getLastName() {
		return lastName;
	}

	public float getFixedSalary() {
		return fixedSalary;
	}

	public Employee getManager() {
		return manager;
	}

	public String getStreet() {
		return street;
	}

	public void setStreet(String street) {
		this.street = street;
	}

	public String getCity() {
		return city;
	}

	public void setCity(String city) {
		this.city = city;
	}

	public String getCountry() {
		return country;
	}

	public void setCountry(String country) {
		this.country = country;
	}

	public float netSalary() {
		float benefits = salaryBenefits();
		float pensionFounds = this.fixedSalary * 10 / 100;
		float tax = 0;
		if (fixedSalary > 3500)
			tax = fixedSalary * 5 / 100;
		return fixedSalary + benefits - pensionFounds - tax;
	}
	
	protected abstract float salaryBenefits();
}

File: src/Net/1. First Course/4. Fourth Hole/RefactoringGolf.Store/Employee.cs
namespace RefactoringGolf.Store
{
    public abstract class Employee
    {
        public string FirstName { get; private set; }

        public string LastName { get; private set; }

        public decimal FixedSalary { get; private set; }

        public Employee Manager { get; internal set; }

        public string Street { get; set; }

        public string City { get; set; }

        public string Country { get; set; }

        protected Employee(string firstName, string lastName, decimal fixedSalary)
        {
            this.FirstName = firstName;
            this.LastName = lastName;
            this.FixedSalary = fixedSalary;
        }

        public decimal NetSalary()
        {
            decimal benefits = SalaryBenefits();
            decimal pensionFounds = this.FixedSalary * 10 / 100;
            decimal tax = 0;
            if (this.FixedSalary > 3500)
                tax = this.FixedSalary * 5 / 100;
            return this.FixedSalary + benefits - pensionFounds - tax;
        }

        protected abstract decimal SalaryBenefits();
    }
}